src/SgidClient.ts

```diff
@@ -243,4 +248,19 @@ export class SgidClient {
     }
     return result
   }
+
+  /**
+   * Parses sgID user data.
+   * @param dataValue A value from the `data` object returned from the `userinfo` method
+   * @returns The parsed data value. If the input is a string, then a string is returned.
+   * If a stringified array or object is passed in, then an array or object is returned
+   * respectively.
+   */
+  parseData(dataValue: string): ParsedSgidDataValue {
+    // JSON parse array data values if necessary
+    if (isStringWrappedInSquareBrackets(dataValue)) {
```

Comment:
Had an offline discussion with Jie Hao whether we should handle isStringWrappedInCurlyBrackets to parse object too.

Reply:
As discussed, we should detect both stringified objects and stringified arrays. Modified in [ff2c39b0](https://github.com/opengovsg/sgid-client/pull/64/commits/ff2c39b01a79f7b156377256f594cc6845033850)